Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/StubIrMetadataEmitter.kt

```diff
@@ -451,39 +451,44 @@ private class MappingExtensions(
      * As of 25 Nov 2019, the latter form is used only for KmTypeAlias.underlyingType.
      */
     // TODO: Add caching if needed.
-    fun StubType.map(shouldExpandTypeAliases: Boolean = true): KmType = when (this) {
-        is AbbreviatedType -> {
-            val typeAliasClassifier = KmClassifier.TypeAlias(abbreviatedClassifier.fqNameSerialized)
-            val typeArguments = typeArguments.map { it.map(shouldExpandTypeAliases) }
-            val abbreviatedType = KmType(flags).also { km ->
-                km.classifier = typeAliasClassifier
-                km.arguments += typeArguments
-            }
-            if (shouldExpandTypeAliases) {
-                // Abbreviated and expanded types have the same nullability.
-                KmType(flags).also { km ->
-                    km.abbreviatedType = abbreviatedType
-                    val kmUnderlyingType = underlyingType.map(true)
-                    km.arguments += kmUnderlyingType.arguments
-                    km.classifier = kmUnderlyingType.classifier
+    fun StubType.map(mode: TypeAliasExpansionMode = TypeAliasExpansionMode.EXPANSION): KmType = when (this) {
```

Comment:
Seems correct from my point of view, but I'm not an expert in metadata.
Have you consulted anyone who is?

Reply:
No, good idea.